integration-cli/docker_utils.go

```diff
@@ -1567,3 +1567,23 @@ func sumAsIntegers(vals ...interface{}) interface{} {
 	}
 	return s
 }
+
+// pollFunc is used to periodically execute a check function, it
+// returns error after timeout.
+func pollFunc(f func() error, timeout time.Duration) error {
+	if f() == nil {
```

Comment:
what's the point of not entering the for with the timer if it returns nil and you exit the for and the function at line 1581? it's a duplicate check then? the function will always exit at that return nil if f() return nil so why not entering the for?